### DISCUSSION TITLE
Not working filer on collections.products

### DISCUSSION THREAD

**Discussion by Yamashou** [Shopify/storefront-api-feedback#88] [Help] [answered]
I want to get results of filtering from collections. products. but I got empty when I set an argument of a filter. 

for example: get that availableForSale is true in collections.

``` graphql
query {
	collections(first: 10) {
			edges {
				node {
					products(
						first: 250
						filters: {
							available: true
						}
					) {
						edges {
							node {
								availableForSale
								productType
						}
					}
				}
			}
		}
	}
}
```

result
```json

{
	"data": {
		"collections": {
			"edges": [
				{
					"node": {
						"products": {
							"edges": []
						}
					}
				},
				{
					"node": {
						"products": {
							"edges": []
						}
					}
				},
				{
					"node": {
						"products": {
							"edges": []
						}
					}
				},
				{
					"node": {
						"products": {
							"edges": []
						}
					}
				},
				{
					"node": {
						"products": {
							"edges": []
						}
					}
				}
			]
		}
	}
}
```
This is an example. But no matter what filter you put in, this result will not change. So I guess not working this argument of filter in collections.products.

**Comment by corygibbons** [comment]
Just double checked a few of my queries and it's working as expected for me. Do you get products back with no filters applied? Maybe double check that both the products in these collections are available on your Storefront API channel and in stock/available?

**Reply by yuzushioh** [reply]
Yes, All the products get back with no filters applied.

**Comment by benzookapi** [comment]
@corygibbons  Hi, I have the same issue with my merchant custom app which responds all product availableForSale = true without filters but once I put availale: true,  the response is empty, and other values of ProductFilter causes empty response so this filters argument seems not working... 

My request id:  6a86d30e-86ec-471d-a860-3a6c6f74a6be

> a few of my queries and it's working as expected for me.


Could you give me your actual queries?

**Comment by yuzushioh** [comment]
This is still broken in the production environment both `2022-01` and `2022-04` . Any updates on this issue? @benzookapi @corygibbons @edward

**Comment by arobbins** [comment]
I'm experiencing the same issue using Storefront API version `2022-04`.

All products from the specified collection are returned, regardless of available status. Here's the query I'm using:

```
query {
  collection(handle: "super") {
    handle
    products(first: 10, filters: [
      {
        available: false
      }
    ]) {
      edges {
        node {
          handle
        }
      }
    }
  }
}
```

**Reply by yuzushioh** [reply]
Thank you for sharing. I'm still having the same issue here.

**Reply by johncraigcole** [reply]
Hey @arobbins, @yuzushioh I just wanted to confirm that while in the Shopify admin, when you goto Online Store-> Navigation, you see a section under Menus for Collection And Search Filters.

**Comment by ksmzn** [comment]
Hi, I have also the same issue and found that `collection.products` doesn't return any `filters` so I can't use available filters.

I remember a few days ago it returned some filters and they were working fine, but now it doesn't.

An example query (This issue happens on `2022-01` and `2022-04` vers both) :

```
query {
  collection(handle: "t-shirt") {
    handle
    products(first: 2) {
      filters {
        id
        label
        type
        values {
          id
          label
          count
          input
        }
      }
      edges {
        node {
          id
          title
        }
      }
    }
  }
}

``` 

Response:

```
{
  "collection": {
    "handle": "t-shirt",
    "products": {
      "filters": [],  // <-- The filters field is always blank.
      "edges": [
        {
          "node": {
            "id": "gid:\/\/shopify\/Product\/< id >",
          }
        },
        :
        :
      ]
    }
  }
}
```

**Comment by benzookapi** [comment]
@johncraigcole  FYI  I attached my shop collection filter setting and add another filter to my query as below.  cc: @yuzushioh 

```
products(
                    first: 1
                    filters: [{
                        price: {
                            min: 0,
                            max: 100000.0
                        }
                    }]
				) {
                    filters {
        id
        label
        type
        values {
          id
          label
          count
          input
        }
      } edges {
						node {
							__typename,
							description,
							productType,
 ......
```

      
     Results:
     
     "products": {


                            "filters": [
                                {
                                    "id": "filter.v.price",
                                    "label": "価格",
                                    "type": "PRICE_RANGE",
                                    "values": [
                                        {
                                            "id": "filter.v.price",
                                            "label": "Price",
                                            "count": 0,
                                            "input": "{\"price\":{\"min\":0,\"max\":0.0}}"
                                        }
                                    ]
                                }
                            ],
                            "edges": []
                        }....


<img width="1033" alt="Screen Shot 2022-04-06 at 11 18 51 AM" src="https://user-images.githubusercontent.com/899580/161883128-bb4495a0-ac13-43a0-affa-2f5f2e3fc4c1.png">

**Reply by johncraigcole** [reply]
Thanks @benzookapi for sharing this. There seems to be an issue with only some filters being available. I'm continuing to investigate and will report back.

**Comment by johncraigcole** [comment, marked as answer]
Hey everyone. 👋 We've confirmed there is an issue with Japanese storefronts and faceted filtering with the Storefront API. I've created an issue to address this internally and can follow up when a resolution is available.

I also wanted to communicate that currently the faceted collection filtering with the Storefront API is tightly coupled to the online store channel. For faceted filtering to work with the Storefront API, products need to be published to both the app channel used for your custom storefront/headless build and the online store channel.

**Comment by tomoyanakano** [comment]
Hi, everyone.
I too faced this issue.

I tried changing my store language to English from Japanese and running query like below.
```graphql
{
    collection(handle: $handle) {
      id
      handle
      title
      products(
        first: $productFirst, 
        filters: $filters,
      ) {
        filters {
          id
          label
          type
        }
      }  
    }
  }
```

```json
Result:

{
  "data": {
    "collection": {
      "products": {
        "filters": [
          {
            "id": "filter.p.product_type",
            "label": "商品タイプ",
            "type": "LIST"
          },
          {
            "id": "filter.v.option.カラー",
            "label": "カラー",
            "type": "LIST"
          },
          {
            "id": "filter.v.option.サイズ",
            "label": "サイズ",
            "type": "LIST"
          },
          {
            "id": "filter.v.availability",
            "label": "出品状況",
            "type": "LIST"
          },
          {
            "id": "filter.v.price",
            "label": "価格",
            "type": "PRICE_RANGE"
          }
        ],
      }
    }
  }
}
```
It is able to fetch available filters and also filter products.
If the language setting is set back to Japanese, the results can still be retrieved as shown above.

Maybe running filter query with English storefront once, this issue would not happen again...🤔

**Comment by Yamashou** [comment]
@johncraigcole 
This issue has not yet been resolved! When will it be resolved? We have been waiting for a year and Merchants are waiting for this feature.

**Comment by tiking76** [comment]
Hi everyone!
I put this app in and it works!
https://apps.shopify.com/search-and-discovery
